leakcanary-android-instrumentation/src/main/java/leakcanary/InstrumentationLeakDetector.kt

```diff
@@ -102,18 +102,21 @@ class InstrumentationLeakDetector {
       return NoAnalysis
     }
 
+    SharkLog.d { "Waiting for idle sync" }
     instrumentation.waitForIdleSync()
     if (!refWatcher.hasWatchedObjects) {
       return NoAnalysis
     }
 
+    SharkLog.d { "Triggering an explicit gc" }
     runGc()
     if (!refWatcher.hasWatchedObjects) {
       return NoAnalysis
```

Comment:
Should we instead add a `reason` field to `NoAnalysis`?

Reply:
After a second thought, I think it would be good to keep both `logging` and `NoAnalysis.reason`. Not all time-consuming steps have a subsequent return statement. For example,
```
    // Giving an extra 2 seconds to flush the hprof to the file system. We've seen several cases
    // of corrupted hprof files and assume this could be a timing issue.
    SharkLog.d { "Waiting for hprof to be flushed" }
    SystemClock.sleep(2000)
```

Reply:
Would know that we're sleeping for 2 seconds help you with your issue?

If you think that's useful we can keep the log statements when there's no return, however they might need to be tightened a bit, e.g instead of `"Triggering an explicit gc"` => add a reason why.

Reply:
Ok, let me first add reasons to `NoAnalysis` as the first step. In my scenario, the function exits immediately after the first GC. If I need more logging, I will do it so as this PR.